Userland/Utilities/sort.cpp

```diff
@@ -56,20 +56,21 @@ struct Options {
     bool unique { false };
     bool numeric { false };
     bool reverse { false };
+    bool zero_terminated { false };
     StringView separator { "\0", 1 };
     Vector<DeprecatedString> files;
 };
 
-static ErrorOr<void> load_file(Options options, StringView filename, Vector<Line>& lines, HashTable<Line>& seen)
+static ErrorOr<void> load_file(Options const& options, StringView filename, StringView line_delimiter, Vector<Line>& lines, HashTable<Line>& seen)
 {
     auto file = TRY(Core::InputBufferedFile::create(
         TRY(Core::File::open_file_or_standard_stream(filename, Core::File::OpenMode::Read))));
 
     // FIXME: Unlimited line length
     auto buffer = TRY(ByteBuffer::create_uninitialized(4096));
     while (TRY(file->can_read_line())) {
-        DeprecatedString line = TRY(file->read_line(buffer));
-
+        StringView line_view { TRY(file->read_until(buffer, line_delimiter)) };
+        DeprecatedString line = line_view;
```

Comment:
I don't think you need to create `line_view`, `DeprecatedString` has a constructor that works:
```c++
DeprecatedString line { TRY(file->read_until(buffer, line_delimiter)) };
```

(It's apparently missing an analogous `operator=`, so that might be the compile error you saw).

Reply:
I think I just forgot to factor this out when I realized I didn't need the variable :sweat_smile: